src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportGenerator;
+import java.util.stream.Collectors;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private static final String FIELDS_NAMES = "fruit,quantity\n";
+
+    @Override
+    public String generateReport() {
+        String builder = Storage.getAllFruits().entrySet().stream()
+                .map(entry -> entry.getKey() + "," + entry.getValue() + "\n")
```

Comment:
Better to use System.lineSeparator() everywhere instead - \n won't work on every OS